src/Lumi/Components/Form/Table.purs

```diff
@@ -0,0 +1,288 @@
+            }
+          ]
+      , validate: \props row ->
+          (f props row).validate
+      }
+  where
+    horizontalRenderer :: Boolean -> Forest JSX -> JSX
+    horizontalRenderer readonly forest =
+      element (R.unsafeCreateDOMComponent "lumi-form")
+        { class: guard readonly "readonly"
+        , children: row_ [ intercalate (hspace S12) (map (toColumn readonly) forest) ]
+        }
+
+    toColumn :: Boolean -> Tree JSX -> JSX
+    toColumn readonly =
+      case _ of
+        Child { key, child } ->
+          maybe identity keyed key $ child
+        Wrapper { key, children } ->
+          R.div
+            { key: fromMaybe "" key
+            , style: R.css { flex: "1" }
+            , children:
+                [ row_
+                    [ intercalate (hspace S12) (map (toColumn readonly) children)
+                    ]
+                ]
+            }
+        Node { key, required: r, validationError, children } ->
+          maybe identity keyed key $
+            Column.column_
+              [ row_
+                  [ redAsterisk r
+                  , intercalate (hspace S12) (map (toColumn readonly) children)
+                  ]
+              , guard (not readonly) $
+                  foldMap errLine validationError
+              ]
+
+    verticalRenderer :: Boolean -> Forest JSX -> JSX
+    verticalRenderer readonly forest =
+      element (R.unsafeCreateDOMComponent "lumi-form")
+        { class: guard readonly "readonly"
+        , children: intercalate (vspace S8) (map (toRow readonly) forest)
+        }
+
+    toRow :: Boolean -> Tree JSX -> JSX
+    toRow readonly =
+      case _ of
+        Child { key, child } ->
+          maybe identity keyed key $ child
+        Wrapper { key, children } ->
+          R.div
+            { key: fromMaybe "" key
+            , children: [ intercalate (vspace S8) (map (toRow readonly) children) ]
+            }
+        Node { key, required: r, validationError, children } ->
+          R.div
+            { key: fromMaybe "" key
+            , children:
+                [ row_
+                    [ redAsterisk r
+                    , Column.column_
+                        [ intercalate (vspace S8) (map (toRow readonly) children)
+                        , guard (not readonly) $
+                            foldMap errLine validationError
+                        ]
+                    ]
+                ]
+            }
+
+    errLine :: ValidationMessage -> JSX
+    errLine =
+      case _ of
+        Error e ->
+          T.text T.subtext
+            { className = Nullable.notNull "labeled-field--validation-error"
+            , children = [ R.text e ]
+            }
+        Warning w ->
+          T.text T.subtext
+            { className = Nullable.notNull "labeled-field--validation-warning"
+            , children = [ R.text w ]
+            }
+
+    redAsterisk :: RequiredField -> JSX
```

Comment:
This feels like it ought to go in the table header?

Reply:
I've based myself on table we currently have which add the red asterisk before the field. But changing this is pretty straightforward if we decide to :)